rdmo/management/viewsets.py

```diff
@@ -110,3 +113,37 @@ def create(self, request, *args, **kwargs):
 
         # step 4: return the list of elements
         return Response(elements)
+
+
+class ElementToggleCurrentSiteViewSetMixin(viewsets.GenericViewSet):
+    permission_classes = (CanToggleElementCurrentSite, )
+    serializer_class = None
+    viewset_class = None
+
+    def get_queryset(self):
+        # needed to pass to the has_permission method
+        viewset_class = self.viewset_class
+        viewset_class.action = self.action
+        return viewset_class.get_queryset(viewset_class)
+
+    @action(detail=True, methods=['put'], url_path="add-site")
```

Comment:
The `permission_class` could go into the `@action` and then the mixin could be used on the main ViewSet. (Not super sure though).